public/containers/RefreshAction/index.tsx

```diff
@@ -0,0 +1,152 @@
+ */
+
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+import { CoreStart } from "opensearch-dashboards/public";
+import { ServicesContext } from "../../services";
+import { CoreServicesContext } from "../../components/core_services";
+import { aliasBlockedPredicate, dataStreamBlockedPredicate, filterBlockedItems, indexBlockedPredicate } from "../../utils/helpers";
+import { IndexOpBlocksType, INDEX_OP_TARGET_TYPE } from "../../utils/constants";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+
+interface RefreshActionModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  onClose: () => void;
+  type: string;
+}
+
+export default function RefreshActionModal<T>(props: RefreshActionModalProps<T>) {
+  const { onClose, visible, selectedItems, type } = props;
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+
+  useEffect(() => {
+    if (!!services && visible) {
+      switch (type) {
+        case INDEX_OP_TARGET_TYPE.INDEX:
+          filterBlockedItems<CatIndex>(services, selectedItems as CatIndex[], IndexOpBlocksType.Closed, indexBlockedPredicate).then(
+            (filteredStreamsResult) => {
+              setUnBlockedItems(filteredStreamsResult.unBlockedItems.map((item) => item.index));
+              setBlockedItems(filteredStreamsResult.blockedItems.map((item) => item.index));
+            }
+          );
+          break;
+        case INDEX_OP_TARGET_TYPE.ALIAS:
+          filterBlockedItems<IAlias>(services, selectedItems as IAlias[], IndexOpBlocksType.Closed, aliasBlockedPredicate).then(
+            (filteredDataStreamsResult) => {
+              setUnBlockedItems(filteredDataStreamsResult.unBlockedItems.map((item) => item.alias));
+              setBlockedItems(filteredDataStreamsResult.blockedItems.map((item) => item.alias));
+            }
+          );
+          break;
+        case INDEX_OP_TARGET_TYPE.DATA_STREAM:
+          filterBlockedItems<DataStream>(
+            services,
+            selectedItems as DataStream[],
+            IndexOpBlocksType.Closed,
+            dataStreamBlockedPredicate
+          ).then((filteredDataStreamsResult) => {
+            setUnBlockedItems(filteredDataStreamsResult.unBlockedItems.map((item) => item.name));
+            setBlockedItems(filteredDataStreamsResult.blockedItems.map((item) => item.name));
+          });
+          break;
+      }
+    } else {
+      setUnBlockedItems([]);
+      setBlockedItems([]);
+    }
+  }, [visible, services]);
```

Comment:
As suzhou suggested, `type` and `selectedItems` should also be included in the dependency.